File: smartdevices-app/smartdevicesapp.model/src/main/java/de/vogler_engineering/smartdevicesapp/model/entities/job/Job.java
/**
 * Copyright (c) Vogler Engineering GmbH. All rights reserved.
 * Licensed under the MIT License. See LICENSE.md in the project root for license information.
 */
package de.vogler_engineering.smartdevicesapp.model.entities.job;

import java.io.Serializable;
import java.util.Map;

import de.vogler_engineering.smartdevicesapp.model.entities.message.Message;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * Created by vh on 14.02.2018.
 */

@Data
@EqualsAndHashCode(callSuper = true)
public class Job extends Message implements Serializable {
    private JobStatus status;
    private boolean immediate;
    private Map<String, String> resource;
}

File: smartdevices-gateway/SmartDevicesGateway.Model/Jobs/Job.cs
//
// Copyright (c) Vogler Engineering GmbH. All rights reserved.
// Licensed under the MIT License. See LICENSE.md in the project root for license information.
//
using System;
using System.Collections.Generic;
using SmartDevicesGateway.Model.Jobs.Enums;
using SmartDevicesGateway.Model.Messages;

namespace SmartDevicesGateway.Model.Jobs
{
    public class Job : Message
    {
        public int LiteDbId { get; set; }

        public Job()
        {
            //for liteDB
        }

        public bool Immediate { get; set; }

        public Dictionary<string, string> Resource { get; set; }

        public JobStatus Status { get; set; }
    }

    public class JobReply : Message
    {
        public string Action { get; set; }

        public Dictionary<string, string> Resource { get; set; }

    }
}